keps/sig-node/4603-tune-crashloopbackoff/README.md

```diff
@@ -846,33 +1099,47 @@ https://storage.googleapis.com/k8s-triage/index.html
+  time stamps to be minimum the backoff, with a healthy timeout
 - k8s.io/kubernetes/test/e2e/node/kubelet_perf: for a given percentage of
 heterogenity between "Succeeded" terminating pods, and crashing pods whose
-`restartPolicy: Always``, 
- * what is the load and rate of Pod restart related API requests to the API
-   server?
- * what are the performance (memory, CPU, and pod start latency) effects on the
-   kubelet component? Considering the effects of different plugins (e.g. CSI,
-   CNI)
+`restartPolicy: Always` or `restartPolicy: OnFailure`, 
+  - what is the load and rate of Pod restart related API requests to the API
+    server?
+  - what are the performance (memory, CPU, and pod start latency) effects on the
+    kubelet component? 
+  - With different plugins (e.g. CSI, CNI)
 
 ### Graduation Criteria
 
 #### Alpha
 
-- Changes to existing backoff curve implemented behind a feature flag 
-    1. Front-loaded decay curve for all workloads with `restartPolicy: Always`
-       <<[UNRESOLVED]>>
-    2. Front-loaded, low max cap backoff curve for nodes workloads with XYZ
-       config
-- New XYZ kubelet config convertable to ABC on downgrade or without feature flag
-<<[/UNRESOLVED]>>
-- Metrics implemented to expose pod and container restart policy statistics,
-  exit states, and runtimes
-- Initial e2e tests completed and enabled
-  * Feature flag on or off
-  * <<[UNRESOLVED]>>node upgrade and downgrade path <<[/UNRESOLVED]>>
-- Fix https://github.com/kubernetes/kubernetes/issues/123602 if this blocks the
-  implementation, otherwise beta criteria
+- New `int32 crashloopbackoff.max` field in `KubeletConfiguration` API, validated
+  to a minimum of 1 and a maximum of 300, used when
+  `EnableKubeletCrashLoopBackoffMax` feature flag enabled, to customize
+  CrashLoopBackOff per node
+- New `ReduceDefaultCrashLoopBackoffDecay` feature flag which, when enabled,
+  changes CrashLoopBackOff behavior (and ONLY the CrashLoopBackOff behavior) to
+  a 2x decay curve starting at 1s initial value, 1 minute maximum backoff for
+  all workloads, therefore affecting workload configured with `restartPolicy:
+  Always` or `restartPolicy: OnFailure`
+    - Requires a refactor of image pull backoff from container restart backoff
+      object
+- Maintain current 10 minute recovery threshold by refactoring backoff counter
+  reset threshold and explicitly implementing container restart backoff behavior
+  at the current 10 minute recovery threshold
+- Initial e2e tests setup and enabled
+- Initial unit tests covering new behavior
+  - Especially confirming the backoff object is set properly depending on the
+feature gates set as per the [Conflict Resolution](#conflict-resolution) policy
+- Test proving `KubeletConfiguration` objects will silently drop unrecognized
+  fields in the `config.validation_test` package
+  ([ref](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/validation/validation_test.go)).
```

Comment:
Is this also the expected behavior when the feature gate is disabled?

Reply:
Yes. I did include this comment inline here in [1515af5](https://github.com/kubernetes/enhancements/pull/4893/commits/1515af5705d87b3f240ae7eb88c7496f0f68bf49)